### PR TITLE
Pop Up Task

### PR DESCRIPTION
**Pull request by SofiiaTrokhymchuk**
### Pop Up Task

[Demo](https://sofiiatrokhymchuk.github.io/hooli-style-pop-up/) |
[Code base](https://github.com/SofiiaTrokhymchuk/hooli-style-pop-up)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css [hunk 1]
```diff
@@ -0,0 +1,217 @@
+* {
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css:1]: 🔥

File: submissions/SofiiaTrokhymchuk/pop-up-task/index.html [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                            <div class="label-wrapper">
+                                <img src="icons/pop-up-icon.svg" alt="Pop-up icon" id="pop-up-icon">
+                            </div>         
+                        </label>  
+                        <div class="pop-up-wrapper">
+                            <div class="pop-up-container">
+                                <ul class="pop-up-menu">
+                                    <li class="pop-up-menu__item">
+                                        <a href="#">
+                                            <figure>
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/index.html:31]: I am not sure that `figure` tag is needed in this case.
Here is it's description. WDYT?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figure

File: submissions/SofiiaTrokhymchuk/pop-up-task/index.html [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                                                <img src="icons/help.svg" alt="Help">
+                                                <figcaption>Help</figcaption>
+                                            </figure>
+                                        </a>    
+                                    </li>
+                                </ul>
+                            </div>
+
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/index.html:115]: remove the empty spaces please

File: submissions/SofiiaTrokhymchuk/pop-up-task/index.html [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                        </div>  
+                        
+                    </li>
+                </ul>            
+            </nav>
+        </header>
+        <main class="main">
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/index.html:122]: 👍

File: submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    max-width: 1000px;
+    margin: 0 auto;
+    position: relative;
+}
+
+.header {
+    display: grid;
+    align-items: center;
+    grid-template-columns: 1fr 2fr;
+    padding: 15px 50px;
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css:37]: what is this trick for? :) I mean padding and margin

File: submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    background-color: #212121;
+    color: #e6e6e6;
+    border-radius: 5px;
+}
+
+.header-menu__item>a:active {
+    color: #f3f3f3;
+}
+
+#pop-up-checkbox:focus~label>div,
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css:68]: better to use spaces for better readability `#pop-up-checkbox: focus ~ label > div`

File: submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+body {
+    font-family: Verdana, Tahoma, sans-serif;
+    color: #212121;
+    background: #e6e6e6;
+    font-size: 1.2rem;
+    height: 100vh;
+}
+
+a:link,
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css:17]: Avoid styling by HTML tag
https://frontstuff.io/you-need-to-stop-targeting-tags-in-css

**Review by kasionio — CHANGES_REQUESTED** [review]
Nice Kottans design 😸 And great job, looking good. 
Please try to fix code according to the comments below. Also `More` button should disappear when you click on it, need to be fixed

File: submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css (hunk 5 above)
Comment (SofiiaTrokhymchuk) [reply submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css:37]: I was playing with paddings and margins inside a container div, and forgot to delete it =(

File: submissions/SofiiaTrokhymchuk/pop-up-task/index.html (hunk 2 above)
Comment (SofiiaTrokhymchuk) [reply submissions/SofiiaTrokhymchuk/pop-up-task/index.html:31]: yep, I agree, that figure was an overhead. thank you!

File: submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css (hunk 5 above)
Comment (SofiiaTrokhymchuk) [reply submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css:37]: I was considering a `.container` as a `div` for all the content on the page. however, I didn't want the `header` to have that padding as well, that's why I used negative margin. but I also want my `header` to have the same padding as a `.container`. without these properties the `header` looks collapsed

**Review by kasionio — CHANGES_REQUESTED** [review]
Nicely done! Please check in git why you have changes of other students in your commit

File: submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+#more-label {
+    cursor: pointer;
+}
+
+#more-checkbox {
+    appearance: none;
+    position: absolute;
+}
+
+#more-checkbox:checked ~ .more-btn{
```
Comment (kasionio) [inline submissions/SofiiaTrokhymchuk/pop-up-task/css/style.css:173]: I guess we can unite it to `.more, #more-checkbox:checked ~ .more-btn { display: none;}`

**Review by kasionio — CHANGES_REQUESTED** [review]
Nicely done!
One suggestion - `More` button doesn't have a focus state when a user using the keyboard.
And take a better look at relative units `rem` and `em`, it's a good practice nowadays. 
You can read articles [here](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units), [here](https://www.freecodecamp.org/news/what-is-rem-in-css/) or any related to it's info

**Comment by SofiiaTrokhymchuk** [comment]
actually, idk where can I use more rem's or em's in my project, because only font-sizes are suitable for this in my case((

**Review by kasionio — APPROVED** [review]
Not only fonts, `max-height`, `padding`... Explore more about relative units, it can be helpful in the future. Anyway, great job here, pretty clean code and good understanding.

**Comment by SofiiaTrokhymchuk** [comment]
thank you so much! I'll definitely explore more about relative units to use them more in my future projects

**Pull request merged**